### PR TITLE
Modernise and fix behavior when used with loading-slider

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Requires https://github.com/discourse/discourse/pull/18491

### CODE REVIEW

**Comment by SamSaffron** [comment]
Is there a "compatibility" thing we can add here so it does not install on old instances?

**Comment by davidtaylorhq** [comment]
Yup, it's already there - `.discourse-compatibility`

File: javascripts/discourse/connectors/bread-crumbs-right/topic-status-filter.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const enabledCategoryIds = settings.enabled_categories
+  .split("|")
+  .filter(Boolean)
+  .map((id) => parseInt(id, 10));
+
+export default {
+  setupComponent(args, component) {
+    // Unfortunately `shouldRender` is not re-evaluated when the arguments change.
```
Comment (CvX) [inline javascripts/discourse/connectors/bread-crumbs-right/topic-status-filter.js:8]: Is this something we'd want to fix/change?

**Review by CvX — APPROVED** [review]

File: javascripts/discourse/connectors/bread-crumbs-right/topic-status-filter.js (hunk 1 above)
Comment (davidtaylorhq) [reply javascripts/discourse/connectors/bread-crumbs-right/topic-status-filter.js:8]: Absolutely, but I didn't want to end up too deep down the rabbit hole this week 😅 

I'd like to run `shouldRender` in an 'autotracking context', so any `@tracked` variables it uses will cause a re-evaluation of the function. Will require some substantial refactoring, but we're gonna need some big changes here anyway to take advantage of glimmer components. 🔜

**Pull request merged**